File: 양승민/입문/src/lec08/lec08Main.java
package lec08;

public class lec08Main {
    public static void main(String[] args) {

        // 4. 같은 타입의 여러 파라미터 받기 (가변인자)
        String[] array = new String[]{"A", "B", "C"};
        printAll(array);

        printAll("A", "B", "C");
    }

    // 1. 함수 선언 문법
    public int max(int a, int b) {
        if (a > b) {
            return a;
        }
        return b;
    }

    // 2. default parameter
    public void repeat(String str, int num, boolean useNewLine) {
        for (int i = 1; i <= num; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }

    public void repeat(String str, int num) {
        repeat(str, num, true);
    }

    public void repeat(String str) {
        repeat(str, 3, true);
    }

    // 4. 같은 타입의 여러 파라미터 받기 (가변인자)
    public static void printAll(String... strings) {
        for (String str : strings) {
            System.out.println(str);
        }
    }

}

File: 양승민/입문/src/lec08/lec08Main.kt
package lec08

fun main() {

    // 1. 함수 선언 문법
    fun max(a: Int, b: Int) = if (a > b) a else b

    // 2. default parameter
    fun repeat(
        str: String,
        num: Int = 3,
        useNewLine: Boolean = true
    ) {
        for (i in 1..num) {
            if (useNewLine) {
                println(str)
            } else {
                print(str)
            }
        }
    }

    // 3. named argument
    repeat("Hello world!", useNewLine = false)

    // 4. 같은 타입의 여러 파라미터 받기 (가변인자)
    fun printAll(vararg strings: String) {
        for (str in strings) {
            println(str)
        }
    }

    val array = arrayOf("A", "B", "C")
    printAll(*array)

    printAll("A", "B", "C")

}